pkg/cluster/config/encoding/encoding.go

```diff
@@ -89,13 +89,20 @@ func Unmarshal(raw []byte) (config.Any, error) {
 	if err != nil {
 		return nil, err
 	}
-	// load version
+	// load version specific config type
 	var cfg config.Any
+	var rawCfg interface{}
 	switch version {
 	case config.SchemeGroupVersion.String():
-		cfg = &config.Config{}
+		// this allows strict unmarshaling with the version header
+		layout := &struct {
+			configHeader
+			config.Config
+		}{}
+		cfg = &layout.Config
+		rawCfg = layout
```

Comment:
sliiightly gross, but it keeps the coupling pretty low, this function is the only thing that needs to know about actual config + the header, just a tiny bit of boilerplate here for each version, and we gain error output on unknown fields